database/clickhouse/clickhouse.go

```diff
@@ -233,19 +241,30 @@ func (ch *ClickHouse) ensureVersionTable() (err error) {
 
 	// if not, create the empty migration table
 	if len(ch.config.ClusterName) > 0 {
-		query = fmt.Sprintf(`
-			CREATE TABLE %s ON CLUSTER %s (
+
+		if ch.config.IsDistributed { // for sharded clusters we want to create a distributed table
+			baseTableName := ch.config.MigrationsTable + "_base"
+			query = fmt.Sprintf(`
+			CREATE TABLE IF NOT EXISTS %s.%s ON CLUSTER %s (
+				version    Int64,
+				dirty      UInt8,
+				sequence   UInt64
+			) Engine=%s Primary Key (sequence)`, ch.config.DatabaseName, baseTableName, ch.config.ClusterName, ch.config.MigrationsTableEngine)
```

Comment:
Should we enforce that the local Engine always be ReplicatedMergeTree. I'm not sure you can create a Distributed table on a standard MergeTree, or that it would make sense for it to be another type.